Microsoft.Azure.CosmosRepository.Samples/WebTier.Integration.Tests/Factories/WebTierApplicationFactory.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Azure.CosmosRepository;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace WebTier.Integration.Tests.Factories
+{
+    public class WebTierApplicationFactory : WebApplicationFactory<Startup>
+    {
+        protected override void ConfigureWebHost(IWebHostBuilder builder)
+        {
+            builder.ConfigureTestServices(services =>
+            {
+                services
+                    .RemoveCosmosRepositories()
```

Comment:
This call is throwing me of, why is it needed?

Reply:
This is because when using the Mvc.Testing package, in the `ConfigureTestServices()` method you have to remove the service descriptors for the existing setup for the real cosmos repository to then be able to replace it with the new in-memory versrion.

Reply:
Had same question. Interesting, thanks for explaining.